src/main/java/org/spongepowered/api/command/Command.java

```diff
@@ -0,0 +1,312 @@
+
+    /**
+     * Test whether this command can probably be executed by the given source.
+     *
+     * <p>If implementations are unsure if the command can be executed by
+     * the source, {@code true} should be returned. Return values of this method
+     * may be used to determine whether this command is listed in command
+     * listings.</p>
+     *
+     * @param source The caller of the command
+     * @return Whether permission is (probably) granted
+     */
+    boolean testPermission(CommandSource source);
+
+    /**
+     * Gets a short one-line description of this command.
+     *
+     * <p>The help system may display the description in the command list.</p>
+     *
+     * @param source The source of the help request
+     * @return A description
+     */
+    Optional<Text> getShortDescription(CommandSource source);
+
+    /**
+     * Gets a longer formatted help message about this command.
+     *
+     * <p>It is recommended to use the default text color and style. Sections
+     * with text actions (e.g. hyperlinks) should be underlined.</p>
+     *
+     * <p>Multi-line messages can be created by separating the lines with
+     * {@code \n}.</p>
+     *
+     * <p>The help system may display this message when a source requests
+     * detailed information about a command.</p>
+     *
+     * @param source The source of the help request
+     * @return A help text
+     */
+    Optional<Text> getHelp(CommandSource source);
+
+    /**
+     * Gets the usage string of this command.
+     *
+     * <p>A usage string may look like
+     * {@code [-w &lt;world&gt;] &lt;var1&gt; &lt;var2&gt;}.</p>
+     *
+     * <p>The string must not contain the command alias.</p>
+     *
+     * @param source The source of the help request
+     * @return A usage string
+     */
+    Text getUsage(CommandSource source);
+
+
+    /**
+     * A high level {@link Builder} for creating a {@link Command}.
+     *
+     * <p>When creating a command, ensure that a {@link CommandExecutor}
+     * <strong>and/or</strong> a child command is specified.</p>
+     */
+    interface Builder extends ResettableBuilder<Command, Builder> {
+
+        /**
+         * Adds a {@link Command} as a child to this command, under the
+         * supplied keys. The keys are case insensitive.
+         *
+         * @param child The {@link Command} that is a child.
+         * @param keys The keys to register as a sub command.
+         * @return This builder, for chaining
+         */
+        Builder addChild(Command child, String... keys);
+
+        /**
+         * Adds a {@link Command} as a child to this command, under the
+         * supplied keys. The keys are case insensitive.
+         *
+         * @param child The {@link Command} that is a child.
+         * @param keys The keys to register as a sub command.
+         * @return This builder, for chaining
+         */
+        Builder addChild(Command child, Iterable<String> keys);
```

Comment:
Builders tend to imply that your always adding on to and up to the build. In that retrospect, should this just be child() (and others like this). addChildren would just be children().